swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,762 @@
+		defer watchCancel()
+
+		log.Info("Setting up stream subscription")
+		// each node Subscribes to each other's swarmChunkServerStreamName
+		var wg sync.WaitGroup
+		for j, id := range ids {
+			log.Trace(fmt.Sprintf("subscribe: %d", j))
+			client, err := net.GetNode(id).Client()
+			if err != nil {
+				return err
+			}
+			wg.Add(1)
+			watchSubscriptionEvents(ctx, id, client, &wg, errc)
+
+			if log.Lvl(*loglevel) >= log.LvlDebug {
+				//uncomment this if to see only the uploader's node kademlia
+				//otherwise print all kademlias
+				//if j == idx {
+				var kt string
+				err = client.CallContext(ctx, &kt, "stream_getKad")
+				if err != nil {
+					return err
+				}
+
+				log.Debug("kad table " + node.ID().String())
+				log.Debug(kt)
+				//}
+			}
+			//watch for peers disconnecting
+			err = streamTesting.WatchDisconnections(id, client, disconnectC, quitC)
+			if err != nil {
+				return err
+			}
+			//start syncing!
+			err = client.CallContext(ctx, nil, "stream_startSyncing")
+			if err != nil {
+				return err
+			}
+		}
+
+		//now wait until the number of expected subscriptions has been finished
+
+		go func() {
+			wg.Wait()
+			close(subscriptionsDone)
+		}()
+
+		select {
+		case <-subscriptionsDone:
+		case err := <-errc:
+			return err
+		}
+		log.Info("Stream subscriptions successfully requested")
+		if live {
+			//now upload the chunks to the selected random single node
+			chunks, err := uploadFileToSingleNodeStore(node.ID(), chunkCount)
+			if err != nil {
+				return err
+			}
+			conf.chunks = append(conf.chunks, chunks...)
+			//finally map chunks to the closest addresses
+			log.Debug(fmt.Sprintf("Uploaded chunks for live syncing: %v", conf.chunks))
+			mapKeysToNodes(conf)
+			log.Info(fmt.Sprintf("Uploaded %d chunks to random single node", chunkCount))
+		}
+
+		//periodically check if chunks have arrived at nodes
+		actionTicker = time.NewTicker(time.Second / 100)
+		conf.metricsWg.Add(1)
+
+		go func() {
+			startTime = time.Now()
+			for range actionTicker.C {
+				checkChunkIsAtNode(conf)
+			}
+		}()
+		log.Info("Action terminated")
+
+		return nil
+	}
+
+	//check defines what will be checked during the test
+	check := func(ctx context.Context, id discover.NodeID) (bool, error) {
+		select {
+		case <-ctx.Done():
+			return false, ctx.Err()
+		case e := <-disconnectC:
+			log.Error(e.Error())
+			return false, fmt.Errorf("Disconnect event detected, network unhealthy")
+		default:
+		}
+		log.Trace(fmt.Sprintf("Checking node: %s", id))
+		//select the local store for the given node
+		lstore := stores[id]
+		//if there are more than one chunk, test only succeeds if all expected chunks are found
+		allSuccess := true
+
+		//all the chunk indexes which are supposed to be found for this node
+		localChunks := conf.idToChunksMap[id]
+		//for each expected chunk, check if it is in the local store
+		for _, ch := range localChunks {
```

Comment:
instead report here percentage of chunks found

Reply:
This will change for every ticker round though, and will finally be 100% if the tests passed. If we want this to be stored over every different ticker round, it needs to be global (hence `checkChunkIsAtNode`)

Reply:
`checkChunkIsAtNode` has been removed from this PR